tests/tests/eth_provider.rs

```diff
@@ -819,6 +831,83 @@ async fn test_send_raw_transaction_wrong_signature(#[future] katana: Katana, _se
 
     // Assert that no transaction is found
     assert!(tx.is_none());
+
+    // Verify that the number of pending transactions in the mempool remains unchanged
+    assert_eq!(eth_provider.mempool().unwrap().pool_size().pending, mempool_size.pending);
+}
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_send_raw_transaction_wrong_chain_id(#[future] katana: Katana, _setup: ()) {
+    // Given
+    let eth_provider = katana.eth_provider();
+    let wrong_chain_id = 999; // An arbitrary wrong chain ID
+
+    // Create a transaction with the wrong chain ID
+    let transaction = Transaction::Eip1559(TxEip1559 {
+        chain_id: wrong_chain_id,
+        nonce: 0,
+        gas_limit: 21000,
+        to: TxKind::Call(Address::random()),
+        value: U256::from(1000),
+        max_fee_per_gas: 875_000_000,
+        max_priority_fee_per_gas: 0,
+        input: Bytes::default(),
+        access_list: Default::default(),
+    });
+
+    // Sign the transaction
+    let signature = sign_message(katana.eoa().private_key(), transaction.signature_hash()).unwrap();
+    let transaction_signed = TransactionSigned::from_transaction_and_signature(transaction, signature);
+
+    // Retrieve the current size of the mempool
+    let mempool_size = eth_provider.mempool().unwrap().pool_size();
+
+    // Attempt to send the transaction
+    let result = eth_provider.send_raw_transaction(transaction_signed.envelope_encoded()).await;
+
+    // Then
+    assert!(result.is_err()); // Ensure the transaction is rejected
+
+    // Verify that the number of pending transactions in the mempool remains unchanged
+    assert_eq!(eth_provider.mempool().unwrap().pool_size().pending, mempool_size.pending);
```

Comment:
I think that here we should assert that the size is zero to really be sure that there is nothing to pollute the pool inside (we called `send_raw_transaction` only once and the conditions were invalid so there should be nothing in the mempool at all)

Reply:
agree, fixed here and in all the other tests